spark/v2.4/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ConstantColumnVector.java

```diff
@@ -119,6 +122,8 @@ public UTF8String getUTF8String(int rowId) {
 
   @Override
   protected ColumnVector getChild(int ordinal) {
-    throw new UnsupportedOperationException("ConstantColumnVector only supports primitives");
+    DataType sparkType = ((StructType) type).fields()[ordinal].dataType();
+    Type childType = SparkSchemaUtil.convert(sparkType);
+    return new ConstantColumnVector(childType, batchSize, ((InternalRow) constant).get(ordinal, sparkType));
```

Comment:
How are these changes related to ORC?

Reply:
I think the code can be removed for now, because previous the ORC constant reading generates a StructReader that needs to be retrieved from the child node, there should be no such problem now.

Reply:
```
ColumnarRow #

  @Override
  public UTF8String getUTF8String(int ordinal) {
    return data.getChild(ordinal).getUTF8String(rowId);
  }
```
To get a column value in a row use `getChild` when performing spark vectorization reads to fetch each column element.

![image](https://user-images.githubusercontent.com/6520673/140884504-fc9eb57c-0183-4713-b0c5-9be0c05fd441.png)